lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_localize_helper.rb

```diff
@@ -299,6 +299,29 @@ def self.merge_xml_documents(all_xmls)
         #
         def self.apply_substitutions(tag)
           tag.content = tag.content.gsub('...', '…')
+
+          # Typography en-dash
+          if tag.content.include?('-')
+            matcher = tag.content.match(/.*(\d+\s*)-(\s*\d+).*/)
```

Comment:
Since the regex is not anchored (`^` and `$`), I think it will only match one occurrence even if the `tag.content` contains multiple ranges. This is also especially because you used `.*` around the regex so it will be greedy on eating all the characters before the last range occurrence and miss all the previous ones.

For example, notice how this only reports a single `MatchData`, and the single match found is for `3-4`, missing `1-2` entirely

```ruby
pry(main)> 'abcd 1-2 def 3-4 ghi'.match(/.*(\d+\s*)-(\s*\d+).*/)
=> #<MatchData "abcd 1-2 def 3-4 ghi" 1:"3" 2:"4">
```

Reply:
Ah! Good catch!